services/director-v2/src/simcore_service_director_v2/modules/service_sidecar/monitor/core.py

```diff
@@ -0,0 +1,394 @@
+        self,
+        service_name: str,
+        node_uuid: str,
+        hostname: str,
+        port: int,
+        service_key: str,
+        service_tag: str,
+        paths_mapping: PathsMappingModel,
+        compose_spec: ComposeSpecModel,
+        target_container: Optional[str],
+        service_sidecar_network_name: str,
+        simcore_traefik_zone: str,
+        service_port: int,
+    ) -> None:
+        """Invoked before the service is started
+
+        Because we do not have all items require to compute the service_name the node_uuid is used to
+        keep track of the service for faster searches.
+        """
+        async with self._lock:
+            if service_name in self._to_monitor:
+                return
+            if node_uuid in self._inverse_search_mapping:
+                raise ServiceSidecarError(
+                    "node_uuids at a global level collided. A running "
+                    f"service for node {node_uuid} already exists. Please checkout "
+                    "other projects which may have this issue."
+                )
+            self._inverse_search_mapping[node_uuid] = service_name
+            self._to_monitor[service_name] = LockWithMonitorData(
+                resource_lock=AsyncResourceLock(False),
+                monitor_data=MonitorData.assemble(
+                    service_name=service_name,
+                    hostname=hostname,
+                    port=port,
+                    service_key=service_key,
+                    service_tag=service_tag,
+                    paths_mapping=paths_mapping,
+                    compose_spec=compose_spec,
+                    target_container=target_container,
+                    service_sidecar_network_name=service_sidecar_network_name,
+                    simcore_traefik_zone=simcore_traefik_zone,
+                    service_port=service_port,
+                ),
+            )
+            logger.debug("Added service '%s' to monitor", service_name)
+
+    async def remove_service_from_monitor(self, node_uuid: str) -> None:
+        # invoked before the service is removed
+        async with self._lock:
+            if node_uuid not in self._inverse_search_mapping:
+                return
+
+            service_name = self._inverse_search_mapping[node_uuid]
+            if service_name not in self._to_monitor:
+                return
+
+            # invoke container cleanup at this point
+            services_sidecar_client: ServiceSidecarClient = get_api_client(self._app)
+
+            # invokes docker compose down, even if the containers are removed they still
+            # seem attached to the network; the network cannot be removed (this is logged
+            # as a failure by the director) and the docker_network will trash the environment
+            # There is no suitable solution to this issue, having netwoks trash
+            # the environment seems to be the best approach :\
```

Comment:
hmm this is not very clear what it means. what means the network trash the environment? can you maybe add an example?

Reply:
I've just checked and this no longer applies. Please ignore the above comment.